app/components/data_protection_confirmation_banner_component.rb

```diff
@@ -50,13 +53,21 @@ def data_protection_officers_text
   def link_text
     if dpo_required?
       "Contact helpdesk to assign a data protection officer"
+    elsif !org_or_user_org.holds_own_stock?
+      "View or add stock owners"
     else
       "Read the Data Sharing Agreement"
     end
   end
 
   def link_href
-    dpo_required? ? GlobalConstants::HELPDESK_URL : data_sharing_agreement_organisation_path(org_or_user_org)
+    if dpo_required?
+      GlobalConstants::HELPDESK_URL
+    elsif !org_or_user_org.holds_own_stock?
```

Comment:
Does this also need an update for if the users org also hasn't signed the agreement?

We be good to have a test for that case as well.

Reply:
Yup, makes sense, thanks! Updated again